src/plugins/unified_histogram/public/container/hooks/use_state_props.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataViewField, DataViewType } from '@kbn/data-views-plugin/common';
+import { getAggregateQueryMode, isOfAggregateQueryType } from '@kbn/es-query';
+import { useCallback, useEffect, useMemo } from 'react';
+import { UnifiedHistogramChartLoadEvent, UnifiedHistogramFetchStatus } from '../../types';
+import type {
+  UnifiedHistogramState,
+  UnifiedHistogramStateService,
+} from '../services/state_service';
+
+export const useStateProps = ({
```

Comment:
I'd try to remove this file and interact with the state service directly. There's some calculation occurring in some of these functions which could be moved into the service as well.

Reply:
I updated this hook to use dedicated state methods which cleans it up a bit, and moved some of the logic into the state service, but I'd like to avoid restructuring the underlying components to use the state service directly -- at least for now until we have clarity on the long term approach for these types of components. It could be worth revisiting in a later PR.